components/feature/downloads/src/main/java/mozilla/components/feature/downloads/DownloadDialogFragment.kt

```diff
@@ -0,0 +1,54 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package mozilla.components.feature.downloads
+
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import mozilla.components.browser.session.Download
+
+typealias OnClickListener = () -> Unit
+
+/**
+ * This is general representation of a dialog meant to be used in collaboration with [DownloadsFeature]
+ * to show a dialog before a download is triggered.
+ * If [SimpleDownloadDialogFragment] is not flexible enough for your use case you should inherit for this class.
+ * Be mindful to call [onStartDownload] when you want to start the download.
+ */
+abstract class DownloadDialogFragment : DialogFragment() {
+
+    /**
+     * A callback to trigger a download, call it when you are ready to start a download. For instance,
+     * a valid use case can be in confirmation dialog, after the positive button is clicked,
+     * this callback must be called.
+     */
+    var onStartDownload: OnClickListener = {}
```

Comment:
The name `OnClickListener` was confusing to me. I was expecting this to be one of Android's OnClickListener classes.

Reply:
Yes, my mistake!
Now, it's fixed